service/service.go

```diff
@@ -0,0 +1,151 @@
+package service
+
+import (
+	"main/repo"
+	"main/transformer"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type ServiceInterface interface {
+	NewRecord(request TransformRequest) *repo.Record
+	GetRecord(id string) (*repo.Record, error)
+	GetRecords() ([]repo.Record, error)
+	UpdateRecord(id string, request TransformRequest) *repo.Record
+	DeleteRecord(id string) error
+}
+
+type service struct {
+	db    DBLayer
+	cache CacheInterface
+}
+
+type DBLayer interface {
+	NewRecord(r *repo.Record) error
+	GetRecord(id string) (repo.Record, error)
+	GetRecords() ([]repo.Record, error)
+	UpdateRecord(r *repo.Record) error
+	DeleteRecord(id string) error
+}
+
+type CacheInterface interface {
+	Set(value *repo.Record)
+	Get(key string) (*repo.Record, bool)
+	Delete(key string)
+}
+
+type TransformRequest struct {
+	Type        string `json:"type"`
+	CaesarShift int    `json:"shift,omitempty"`
+	Input       string `json:"input,omitempty"`
+}
+
+func NewService(db DBLayer, cache CacheInterface) service {
+	return service{
+		db:    db,
+		cache: cache,
+	}
+}
+
+func (s service) NewRecord(request TransformRequest) *repo.Record {
```

Comment:
let's rename `NewRecord` to `CreateRecord`, because `New` stands more for the constructor rather than save operation